src/main/JayZ.ts

```diff
@@ -1,49 +1,119 @@
-import { EncryptedJayZItem, Encryptor } from "./encryptors"
+import { memzero, ready } from "libsodium-wrappers"
+import { DataKey, DataKeyProvider } from "./DataKeyProvider"
+import { Encryptor } from "./Encryptor"
+import { LibsodiumEncryptor } from "./LibsodiumEncryptor"
+import { EncryptedItemMetadata, EncryptedJayZItem } from "./types"
 
-export interface JayZProps {
-  encryptor: Encryptor
+export type JayZConfig = {
+  keyProvider: DataKeyProvider
+  encryptor?: Encryptor
+  maxUsesPerDataKey?: number
 }
 
-export interface EncryptItemProps<T, U extends keyof T> {
+export type EncryptItemRequest<T, K extends keyof T> = {
   item: T
-  fieldsToEncrypt: U[]
+  fieldsToEncrypt: K[]
 }
 
 export class JayZ {
-  private encryptor: Encryptor
+  public readonly ready = ready
```

Comment:
This was likely added to give callers a convenient way to determine if the JayZ class could be used. For context, the libsodium library is compiled to JS from C and uses WASM, so you can't use it until this `ready` promise is completed.